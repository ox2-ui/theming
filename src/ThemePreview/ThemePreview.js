import React from 'react';

const styles = {
  wrapper: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  box: {
    margin: '10px',
    textAlign: 'center',
  },
  swatch: {
    width: '30px',
    height: '30px',
    margin: '0 auto',
  },
  label: {
    color: 'grey',
    height: '50px',
    marginTop: '5px',
  },
};

const colors = [
  // UI Element colors
  'white',
  'item',
  'backdrop',
  'backdrop-tint',
  'item-alt',
  'backdrop-alt',
  // Neutral colors
  'neutral-tint',
  'neutral-faded',
  'neutral',
  'neutral-light',
  'neutral-dim',
  'neutral-dark',
  'dark',
  'light',
  'light-faded',
  // Status colors
  'action',
  'warning',
  'attention',
  'critical',
  'positive',
  'negative',
  'highlight',
  // Branding colors
  'brand',
  'primary-header',
  'action-menu',
  'action-menu-items',
  'main-menu-header',
  'main-menu-items',
  'main-menu-button',
  // Social network colors
  'twitter',
  'facebook',
  'linkedin',
  'google',
];

/**
 * Default theme for docs and playground
 */
const ThemePreview = () => {
  return (
    <div>
      Available color names:
      <div style={styles.wrapper}>
        {colors.map((item, i) => {
          return (
            <div key={i} style={styles.box}>
              <div
                className={`color:${item}`}
                style={styles.swatch}
              />
              <div style={styles.label}>{item}</div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default ThemePreview;
