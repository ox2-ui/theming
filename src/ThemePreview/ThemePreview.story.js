import React from 'react';
import { storiesOf } from '@kadira/storybook';
import ThemePreview from './ThemePreview';
import ThemeLoader from '../ThemeLoader';
import { default as theme } from '@tapfuse/theme-spark';

storiesOf('ThemePreview', module).add('theme spark', () => (
  <div>
    <ThemeLoader theme={theme} />
    <ThemePreview />
  </div>
));
