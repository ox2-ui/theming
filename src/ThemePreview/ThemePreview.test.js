/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import ThemePreview from './ThemePreview';

const handler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <ThemePreview />
  );

  expect(wrapper).toMatchSnapshot();
});
