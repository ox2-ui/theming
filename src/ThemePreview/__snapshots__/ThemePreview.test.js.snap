// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ThemePreview>
  <div>
    Available color names:
    <div
      style={
        Object {
          "display": "flex",
          "flexWrap": "wrap",
        }
      }
    >
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:white"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          white
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:item"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          item
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:backdrop"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          backdrop
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:backdrop-tint"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          backdrop-tint
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:item-alt"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          item-alt
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:backdrop-alt"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          backdrop-alt
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:neutral-tint"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          neutral-tint
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:neutral-faded"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          neutral-faded
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:neutral"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          neutral
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:neutral-light"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          neutral-light
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:neutral-dim"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          neutral-dim
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:neutral-dark"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          neutral-dark
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:dark"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          dark
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:light"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          light
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:light-faded"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          light-faded
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:action"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          action
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:warning"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          warning
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:attention"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          attention
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:critical"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          critical
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:positive"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          positive
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:negative"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          negative
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:highlight"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          highlight
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:brand"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          brand
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:primary-header"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          primary-header
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:action-menu"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          action-menu
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:action-menu-items"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          action-menu-items
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:main-menu-header"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          main-menu-header
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:main-menu-items"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          main-menu-items
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:main-menu-button"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          main-menu-button
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:twitter"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          twitter
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:facebook"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          facebook
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:linkedin"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          linkedin
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "10px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="color:google"
          style={
            Object {
              "height": "30px",
              "margin": "0 auto",
              "width": "30px",
            }
          }
        />
        <div
          style={
            Object {
              "color": "grey",
              "height": "50px",
              "marginTop": "5px",
            }
          }
        >
          google
        </div>
      </div>
    </div>
  </div>
</ThemePreview>
`;
